src/tests/curve.rs

```diff
@@ -381,3 +387,45 @@ pub fn svdw_map_to_curve_test<G: CurveExt>(
         assert_eq!(output, expected);
     }
 }
+
+macro_rules! curve_testing_suite {
+    ($($curve: ident),*) => {
+
+        #[cfg(test)]
+        mod tests {
+            use super::*;
+
+            #[test]
+            fn test_curve() {
+                $(
+                    crate::tests::curve::curve_tests::<$curve>();
```

Comment:
Instead of coding the functions generic and then calling them here in the macro, why don't you code the things in the macro directly?

On this way, you have a lot less `pub` functions needed and visible across the crate (only the macro).

This applies to all the tests that look like this.

Reply:
Thanks for the comment! 
I updated the curve testing macro as you recommended. https://github.com/privacy-scaling-explorations/halo2curves/pull/129/commits/7c8501d6a6686c071025633e9634d3ce309c4c5f
Will try the same for field testing macro.